R/models.R

```diff
@@ -0,0 +1,261 @@
+  }
+}
+
+
+
+
+
+
+
+#' @title Get Model Details
+#'
+#' @description
+#' \code{get_model_details} Calls Model Details endpoint of API
+#'
+#' @details
+#' Every model has its own parameters and criteria it uses to calculate the results.
+#' This API will return those details; for the growth stage models currently available
+#' this API returns information about the crop stages that the model may return.
+#'
+#' @references https://developer.awhere.com/api/reference/models/details
+#'
+#' @param - model_id: A model id to retrieve details for that specific model
+#' @param - keyToUse: aWhere API key to use.  For advanced use only.  Most users will not need to use this parameter (optional)
+#' @param - secretToUse: aWhere API secret to use.  For advanced use only.  Most users will not need to use this parameter (optional)
+#' @param - tokenToUse: aWhere API token to use.  For advanced use only.  Most users will not need to use this parameter (optional)
+#'
+#' @import httr
+#'
+#' @return - data.frame containing details about requested model
+#'
+#' @examples
+#' \dontrun{get_model_details('BarleyGenericMSU')
+#' }
+
+#' @export
+
+get_model_details <- function(model_id
+                      ,keyToUse = awhereEnv75247$uid
+                      ,secretToUse = awhereEnv75247$secret
+                      ,tokenToUse = awhereEnv75247$token) {
+  
+  checkCredentials(keyToUse,secretToUse,tokenToUse)
+  
+  ## Create Request
+  url <- "https://api.awhere.com/v2/agronomics/models/"
+  
+  url <- paste0(url, model_id, "/details")
+  
+  
+  doWeatherGet <- TRUE
+  while (doWeatherGet == TRUE) {
+    request <- httr::GET(url,
+                         httr::content_type('application/json'),
+                         httr::add_headers(Authorization =
+                                             paste0("Bearer ", tokenToUse)))
+    
+    a <- suppressMessages(httr::content(request))
+    
+    doWeatherGet <- aWhereAPI:::check_JSON(a,request)[[1]]
+  }
+  
+  
+  a[sapply(a, is.null)] <- NA
+  
+  data <- data.frame(biofix = a$biofix, gddMethod = as.character(a$gddMethod), gddBaseTemp = a$gddBaseTemp, gddMaxBoundary = as.integer(a$gddMaxBoundary),
+                     gddMinBoundary = as.integer(a$gddMinBoundary), gddUnits = as.character(a$gddUnits))
+  
+  stages <- as.data.frame(do.call(rbind, lapply(a$stages, rbind)))[, c(2:5)]
+  stages <- data.frame(lapply(stages, as.character), stringsAsFactors=FALSE)
+  stages$gddThreshold <- as.integer(stages$gddThreshold)
+  
+  data <- cbind(data, stages)
+  colnames(data) <- c("biofix", "gddMethod", "gddBaseTemp", "gddMaxBoundary", "gddMinBoundary",
+                      "gddUnits", "id", "stage", "description", "gddThreshold")
+  rownames(data) <- c(1:nrow(data))
+  
+  data <- dplyr::mutate_if(data, is.factor, as.character)
+  
+  
+  if(nrow(data) == 0) {
+    stop(a$simpleMessage)
+  } else {
+    return(as.data.frame(data))
+  }
+}
+
+
+
+
+
+
+#' @title Get Model Results
```

Comment:
This doesn't work for me.  I get back a different named list in "a" then you are expecting when I use this code below
[credentials.txt](https://github.com/aWhereAPI/aWhere-R-Library/files/3560341/credentials.txt)


library(aWhereAPI)

load_credentials('c:/aWhere/credentials/credentials.txt')

get_fields()
get_planting('field_test4')

get_model_results(field_id = 'field_test4'
                  ,model_id = 'Corn2500ISUAbendroth')